src/main.rs

```diff
@@ -164,18 +164,22 @@ fn main() -> std::io::Result<()> {
 }
 
 fn write_once(buffer: &[u8]) -> std::io::Result<Duration> {
-    let mut write_time = Duration::new(0, 0);
+    let start = Instant::now();
```

Comment:
The idea was to measure only write relevant stuff, but not progress update stuff. This is why it was a Duration that sums up over time.